src/lsp.rs

```diff
@@ -70,6 +70,11 @@ impl Client {
 					"Content-Length" => {
 						content_len = sp[1].parse().unwrap();
 					}
+					"Content-Type" => {
+						if sp[1].contains("application/json") {
+							panic!("content-type header set but doesn't contain application/json, {}", sp[1]);
+						}
```

Comment:
The langserver protocol docs say the default is `application/vscode-jsonrpc; charset=utf-8`, and this code is checking for the existence of `application/json`, which will cause it to fail (and which explains why it works on your machine: I suspect the ocaml lsp server is returning the vscode application type, not json).

```suggestion
						if sp[1] != "application/vscode-jsonrpc; charset=utf-8" {
							panic!("unexpected content-type: {}", sp[1]);
						}
```

Reply:
Can you try this and tell me if it works with ocaml lsp?

Reply:
Sorry, didn't realise that you'd replied, ```% ./target/release/acre
thread '<unnamed>' panicked at 'unexpected content-type: application/vscode-jsonrpc; charset=utf-8', src/lsp.rs:75:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
read event err: deleted window
Error: receiving on an empty and disconnected channel
% ``` guess not

Reply:
That's a real confusing error message because it matches exactly the string we are match on with `!=`. Are you sure you copied my code suggestion exactly?

Reply:
I've never used ocaml, but am trying to figure this out locally.

Reply:
Yeah seems to be mistake on my end, apologies